pkg/client/README.md

```diff
@@ -0,0 +1,132 @@
+# William
+
+This library aims to prevent writing to much code using william on your application.
+
+## Creating a service
+```go
+wi := william.New("URL OF WILLIAM", "ServiceName")
+```
+You can set the HTTP Client that will made the request also the tracing if anyone. (if not set nil)
+```go
+wi.SetClient(
+    &http.Client{
+        Timeout: time.Second * 2,
+    },
+    nil,
+)
+```
+
+If your service will use any internal function of william like a service, Ej: adding permission directly from your service and not from the Hub you need to set a KeyPair to auth.
+
+```go
+wi.SetKey(
+    "ServiceKeyID",
+    "ServiceKeySecret",
+)
+```
+
+By default William is created with a bypass, you can enable with:
+```go
+wi.Enable()
+```
+
+## Permissions Handler
+You want to validate if a user have a valid permission for a specific Handler. If your application don't care about resource you can use william.Generate("RL", "Action") that will wrap this for you.
+```go
+mux := http.NewServeMux()
+mux.HandleFunc("/action",
+    wi.HandlerFunc(wi.Generate("RL", "Action"),
+    func(w http.ResponseWriter, r *http.Request) {
+        // On Request you will have access to william.
+        auth := william.Auth(r)
+        fmt.Fprintf(w, "Your mail is: %s!", auth.Email())
+    }),
+)
+```
+I recommend always to register you actions using the method by your william instance, so you can have a generated list for /am.
+
+If you want to create a custom permission check, provide a function like this: 
```

Comment:
I'm not sure I understood this scenario. It seems you are creating the permission by hand, but what would be a `custom permission check`? How would this function be used?

Reply:
This is more related to the Gorrila Mux example If your permission needs a dynamic pattern from the request. Ex: check the permission for a specific resource, you need to provide a string with that pattern.